content/learn/08.contributions/00.arduino-writing-style-guide/arduino-writing-style-guide.md

```diff
@@ -59,7 +59,7 @@ if (somethingIsTrue) doSomething;
 Instead, use this:
 
 ```
-if (somethingIsTrue == TRUE) {
+if (somethingIsTrue) {
```

Comment:
```suggestion
if (somethingIsTrue == true) {
```

Reply:
Hey @jfjlaros , I suggest that we keep the `== true` in this if statement, as for less experienced coders, it describes what is happening a bit better.

While `if(somethingIsTrue){}` works the same and involves less code, it is harder to understand for a beginner, as the state it is compared (true) is not clear. If you want to, we can add this as a paragraph below.

Thanks a lot for your submission, let me know what you think.

Reply:
I understand your concerns, but I am of the opinion that we should not encourage this practice. As mentioned in the C++ core guidelines, it is redundant and opens opportunities for errors. The current issue is a good example of this.

I also think that adding redundant truth tests will get the wrong message across for beginners, as they may get the impression that `if` is some sort of function that takes two operands.

As a compromise, maybe the example could be changed to one that uses a meaningful comparison, e.g.,

```cpp
if (distance > 10) {
  // Go nearer.
}
```

Then we could leave the choice of using `== true` up to the author of the tutorial.